lib/ansible/cli/__init__.py

```diff
@@ -218,9 +229,9 @@ def build_vault_ids(vault_ids, vault_password_files=None,
         return vault_ids
 
     @staticmethod
-    def setup_vault_secrets(loader, vault_ids, vault_password_files=None,
-                            ask_vault_pass=None, create_new_password=False,
-                            auto_prompt=True):
+    def setup_vault_secrets(loader: DataLoader, vault_ids: list[Any], vault_password_files: Optional[list[Any]] = None,
+                            ask_vault_pass: Optional[bool] = None, create_new_password: bool = False,
```

Comment:
`Optional` is considered confusing. There's even linters/formatters for replacing `Optional[Thing]` with `Union[Thing, None]`. Though, with `from __future__ import annotations` above, it's even clearer with the newer syntax:
```suggestion
                            ask_vault_pass: bool | None = None, create_new_password: bool = False,
```